gapis/api/vulkan/api/queue.api

```diff
@@ -137,140 +143,57 @@ cmd VkResult vkQueueSubmit(
           wait_semaphores_all_valid.b = false
           vkErrorInvalidSemaphore(ws)
         } else {
-          semaphore := Semaphores[ws]
+          subm.WaitSemaphores[len(subm.WaitSemaphores)] = ws
+        }
+      }
+    }
 
-          if(semaphore.SubmitCount == 0) {
-            vkErrorSemaphoreNotSubmitted(ws)
```

Comment:
We had a case that sometimes we get waits on semaphores without they are being submitted. This error was to prevent the deadlock and report it. I see that this is being removed. Are we sure we are not reintroducing the bug?

Reply:
Although I see that we don't have pending commands anymore.

Reply:
With this change, no commands ever are put in the pending state. So we now simply don't care what state a semaphore is in (from a simulation perspective).